cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/artifact/RemotePluginFinder.java

```diff
@@ -167,17 +167,13 @@ private List<PluginInfo> getPlugins(NamespaceId namespaceId,
 
     int responseCode = response.getResponseCode();
     if (responseCode != HttpURLConnection.HTTP_OK) {
-      switch (responseCode) {
-        // throw retryable error if app fabric is not available, might be due to restarting
-        case HttpURLConnection.HTTP_BAD_GATEWAY:
-        case HttpURLConnection.HTTP_UNAVAILABLE:
-        case HttpURLConnection.HTTP_GATEWAY_TIMEOUT:
-          throw new ServiceUnavailableException(
-            Constants.Service.APP_FABRIC_HTTP,
-            Constants.Service.APP_FABRIC_HTTP + " service is not available with status " + responseCode);
-        case HttpURLConnection.HTTP_NOT_FOUND:
-          throw new PluginNotExistsException(namespaceId, pluginType, pluginName);
-
+      if (io.cdap.cdap.api.common.Constants.HTTP_SERVER_ERROR_CODES.contains(responseCode)) {
```

Comment:
in what situation are we getting a 500 response? Is it supposed to be returning a different error code? It doesn't seem correct to treat a 500 as if the service is unavailable, 500 normally means we were able to contact the server but some other error happened.

Reply:
We actually haven't seen 500 response returned in RemotePluginFinder - I just updated this code so that the behavior is consistent with other callers.

Actual issue observed: IVP returns 500 when we make a request using RemoteClient from HDF to CDF.
Regarding retrying on 500 response, here's the internal guidance from go/canonical-codes for service implementers:
```
Use UNAVAILABLE if the client can retry just the failing call.
```
From a comment in code.proto (in google3):
```
// The service is currently unavailable.  This is most likely a
// transient condition, which can be corrected by retrying with
// a backoff.
```

So it seems 500 returned (atleast by google services like IVP) can be retried.

Reply:
Looking at docs for some google services, recommendation is to to retry on 500 response.

[PubSub](https://cloud.google.com/pubsub/docs/reference/error-codes) - The client can retry the operation.
[BigQuery](https://cloud.google.com/bigquery/docs/error-messages) - Wait according to the back-off requirements described in the [BigQuery Service Level Agreement](https://cloud.google.com/bigquery/sla), then try the operation again.